### DISCUSSION TITLE
Is it possible a direct call to fit cropping bbox to the centre of screen while clicking a Fit-In-Window button?

### DISCUSSION THREAD

**Discussion by AlbertLiu1974** [CropperBlazor/Cropper.Blazor#323] [General]
Greetings, It seems I need a similar function after copping some images in the list and try to recall when clicking the fit-in button. Kindly refer to the image blow, thanks a lot. 

![image](https://github.com/CropperBlazor/Cropper.Blazor/assets/152601781/1db1b504-bd28-40d5-b38e-f3d1a3cade6d)

Best regards,
Albert Liu

**Comment by MaxymGorn** [comment]
Hi @AlbertLiu1974, as I understood correctly, you want to select a specific coordinate of the cropping box on the image?

**Reply by MaxymGorn** [reply]
If yes, this function (get and set Crop Box Data) is avaliable in our demo example and API library: https://cropperblazor.github.io/demo

**Reply by AlbertLiu1974** [reply]
Hi @MaxymGorn , 

  Yes. I followed the code you provide. Kind of working. Below is the code snippet: maybe not a good one but works.

//move bounding box and image to the origin point 0,0
 ImageData currentImageData = await cropperRef!.GetImageDataAsync();
 decimal currentZoomRatio = currentImageData.Width / currentImageData.NaturalWidth;
ContainerData containerData = await cropperRef!.GetContainerDataAsync();
cropperRef?.ZoomTo(1, containerData.Width / 2, containerData.Height / 2);
cropperRef?.MoveTo(0, 0);

cropX = rowSelectableEventArgs.Item?.Points?[0][0];
cropY = rowSelectableEventArgs.Item?.Points?[0][1];
cropWidth = rowSelectableEventArgs.Item?.Points?[1][0] - cropX;
cropHight = rowSelectableEventArgs.Item?.Points?[2][1] - cropY;

//move bbox
 cropperRef?.SetData(new SetDataOptions()
     {
         X = cropX,
         Y = cropY,
         Height = cropHight,
         Width = cropWidth,
         Rotate = 0,
         ScaleX = 1,
         ScaleY = 1
     });

//calculate bounding box to the center of screen. an offset for bbox and image
 int newCenterX = imgWidth / 2;
 int newCenterY = imgHeight / 2;
int? deltaX = newCenterX - (cropX + (cropWidth / 2));
int? deltaY = newCenterY - (cropY + (cropHight / 2));
 cropperRefDst?.Move(decimal.Parse(deltaX?.ToString()), deltaY);
 cropperRefDst?.SetData(new SetDataOptions()
     {
         X = (cropX),
         Y = (cropY),
         Height = cropHight,
         Width = cropWidth,
         Rotate = 0,
         ScaleX = 1,
         ScaleY = 1
     });

Best regards,
Albert Liu

**Reply by AlbertLiu1974** [reply]
But somehow when I change the size to larger.. the behavior strange..

![image](https://github.com/CropperBlazor/Cropper.Blazor/assets/152601781/fcfa176c-12d3-4139-a283-2a1ff97f9d52)

Sounds the coordinate is wrong when zoom and fit..

Br, Albert

**Reply by AlbertLiu1974** [reply]
int newCenterX = imgWidth / 2;
int newCenterY = imgHeight / 2;

--> Change imgWidth and imgHeight to containerData resolve this problem.  kind of img-container or big-img style effected. Just curious the container data size factor. Sounds MAX of each browser or screen. 

Br, Albert
